x/gov/legacy/v040/types.go

```diff
@@ -0,0 +1,302 @@
+// Package v040 is take from:
+// https://github.com/cosmos/cosmos-sdk/blob/v0.41.1/x/gov/types/vote.go
+// by copy-pasted only the relevants parts for Vote.
```

Comment:
In v040 we had

```go
type Vote struct {
	ProposalId uint64           `protobuf:"varint,1,opt,name=proposal_id,json=proposalId,proto3" json:"proposal_id,omitempty" yaml:"proposal_id"`
	Voter      string           `protobuf:"bytes,2,opt,name=voter,proto3" json:"voter,omitempty"`
	Option     types.VoteOption `protobuf:"varint,3,opt,name=option,proto3,enum=cosmos.gov.v1beta1.VoteOption" json:"option,omitempty"`
}
```

and in v042 we add a new field, and marked the 3rd field as reserved, which gives

```go
type Vote struct {
	ProposalId uint64               `protobuf:"varint,1,opt,name=proposal_id,json=proposalId,proto3" json:"proposal_id,omitempty" yaml:"proposal_id"`
	Voter      string               `protobuf:"bytes,2,opt,name=voter,proto3" json:"voter,omitempty"`
	Options    []WeightedVoteOption `protobuf:"bytes,4,rep,name=options,proto3" json:"options"`
}
```

Here, I went through the old .pb.go and copy-pasted the minimum to have the old vote struct and its proto marshaling working.